### DISCUSSION TITLE
How can I use commands to replace a bad hard drive when the my zpool is offline and cannot be imported

### DISCUSSION THREAD

**Discussion by gesangtome** [openzfs/zfs#16182] [Q&A]
Hello, everybody.

I have a RAID-Z1 pool that uses 5 hard drives, and I recently received a SMART alert about 1 of them. When I shut down my NAS, found the bad hard drive and inserted a new one, then turned it back on and executed the zfs replace command, zfs reminded me that the pool didn't exist.

root@truenas[/home/admin]# zpool import massstorage
cannot import 'massstorage': I/O error
        Destroy and re-create the pool from
        a backup source.
root@truenas[/home/admin]# 

It looks like zfs can only be replaced in an imported pool, can I execute the replacement command on an offline ZFS pool?  What can I do to get ZFS to automatically rebuild my RAID-Z1 pool, which has about 6TB of precious data in it.

Pious need help from anyone  :)

**Comment by IvanVolosyuk** [comment]
To replace a failed disk you have to have enough redundancy left to import the pool and do the replacement. If two of the disks failed for raidz1 you may not have enough disks to recover the data and the pool will fail to import. I would start by identifying where your data is - which disks or partitions hold the data. Use the following command with one of the partitions constituting the zpool:
zdb -l /dev/sdXX 
This will dump the label of the disk telling you which disks it expects to see to import. Make sure it mentions ```type: 'raidz'```. You can try to import manually specifying the disk/partitions which constitute the pool in no particular order: ```zpool import -d /dev/xxx ....``` or ```zpool import -d /dev/disk/by-id```

**Reply by gesangtome** [reply]
Hello

The entire Pool consists of 5*3T Seagate hard drives with only one data dev, which I tried to import and output

root@truenas[/home/admin]# zpool import -d /dev/disk/by-id                            pool: massstorage
     id: 18149558291568110212
  state: FAULTED
status: One or more devices contains corrupted data.
 action: The pool cannot be imported due to damaged devices or data.
        The pool may be active on another system, but can be imported using
        the '-f' flag.
   see: https://openzfs.github.io/openzfs-docs/msg/ZFS-8000-5E
 config:

        massstorage                                         FAULTED  corrupted data          raidz1-0                                          DEGRADED
            c364addd-2e9c-ba41-9fde-264ce3ec7608            UNAVAIL
            usb-ST3000VX_010-2E3166_202105071076-0:3-part1  ONLINE
            usb-ST3000VX_010-2E3166_202105071076-0:2-part1  ONLINE
            usb-ST3000VX_010-2E3166_202105071076-0:1-part1  ONLINE
            usb-ST3000VX_010-2E3166_202105071076-0:0-part1  ONLINE
root@truenas[/home/admin]#

The four disk device files can be found under /dev

find /dev/disk/by-id/ -name "usb-ST3000*"
/dev/disk/by-id/usb-ST3000VX_010-2E3166_202105071076-0:2
/dev/disk/by-id/usb-ST3000VX_010-2E3166_202105071076-0:0
/dev/disk/by-id/usb-ST3000VX_010-2E3166_202105071076-0:3
/dev/disk/by-id/usb-ST3000VX_010-2E3166_202105071076-0:1
/dev/disk/by-id/usb-ST3000VX_010-2E3166_202105071076-0:1-part9
/dev/disk/by-id/usb-ST3000VX_010-2E3166_202105071076-0:3-part9
/dev/disk/by-id/usb-ST3000VX_010-2E3166_202105071076-0:3-part1
/dev/disk/by-id/usb-ST3000VX_010-2E3166_202105071076-0:0-part9
/dev/disk/by-id/usb-ST3000VX_010-2E3166_202105071076-0:0-part1
/dev/disk/by-id/usb-ST3000VX_010-2E3166_202105071076-0:2-part1
/dev/disk/by-id/usb-ST3000VX_010-2E3166_202105071076-0:2-part9
/dev/disk/by-id/usb-ST3000VX_010-2E3166_202105071076-0:1-part1
root@truenas[/home/admin]#

**Comment by lundman** [comment]
it's possible you removed wrong disk and now it thinks it has lost 2 and can't proceed. restore, and run a "zpool import" so we can see how many are bad

**Reply by gesangtome** [reply]
Hello

thank you for your reply, I checked my disk again and was sure that only one piece was missing, when I import the pool, output these


root@truenas[/home/admin]# zpool import massstorage

   pool: massstorage
     id: 18149558291568110212
  state: FAULTED
status: One or more devices contains corrupted data.
 action: The pool cannot be imported due to damaged devices or data.
        The pool may be active on another system, but can be imported using
        the '-f' flag.
   see: https://openzfs.github.io/openzfs-docs/msg/ZFS-8000-5E
 config:

        massstorage                               FAULTED  corrupted data
          raidz1-0                                DEGRADED
            c364addd-2e9c-ba41-9fde-264ce3ec7608  UNAVAIL
            fad28274-1fae-c54a-96ae-f8bfb5286566  ONLINE
            736b4ae1-6776-bd44-8a0f-2542a439f4e5  ONLINE
            cbc8c852-10a0-314b-8a05-a038d5ae3e4c  ONLINE
            b278d068-02be-ee43-9f69-32a433457eec  ONLINE
root@truenas[/home/admin]#

The remaining four disks appear to be online
